drivers/modem/gsm_ppp.c

```diff
@@ -788,7 +790,7 @@ static int gsm_init(const struct device *dev)
 	gsm->gsm_data.rx_rb_buf_len = sizeof(gsm->gsm_rx_rb_buf);
 
 	r = modem_iface_uart_init(&gsm->context.iface, &gsm->gsm_data,
-				  CONFIG_MODEM_GSM_UART_NAME);
+				  GSM_UART_DEV_NAME);
```

Comment:
This one is used by a few more modem drivers, but if you don't mind converting it to take a const struct device while you're here, I think that would be a nicer result.